components/director/internal/open_resource_discovery/ord_document.go

```diff
@@ -280,15 +307,18 @@ func (docs Documents) Validate(calculatedBaseURL string, apisFromDB map[string]*
 				}
 			}
 		}
+
+		doc.APIResources = deleteInvalidInputObjects(invalidApisIndices, doc.APIResources)
+		doc.EventResources = deleteInvalidInputObjects(invalidEventIndices, doc.EventResources)
 	}
```

Comment:
should we do `invalidApisIndices = nil` and  `invalidEventIndices = nil` here as well?

Reply:
Yes, we should, because if they aren't there, the indices from previous iterations will be stored in the next ones. Thanks for the comment  I added the clearing of the slices.